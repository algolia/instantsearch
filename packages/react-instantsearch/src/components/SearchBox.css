.root {
  display: inline-block;
  position: relative;
  width: 500px;
  height: 41px;
  white-space: nowrap;
  box-sizing: border-box;
  font-size: 14px;
}

.wrapper {
  width: 100%;
  height: 100%;
}

.input {
  display: inline-block;
  transition: box-shadow .4s ease, background .4s ease;
  border: 0;
  border-radius: 4px;
  box-shadow: inset 0 0 0 1px #CCCCCC;
  background: #FFFFFF;
  padding: 0;
  padding-right: 77px;
  padding-left: 11px;
  width: 100%;
  height: 100%;
  vertical-align: middle;
  white-space: normal;
  font-size: inherit;
  appearance: none;
}

.input::-webkit-search-decoration, .input::-webkit-search-cancel-button, .input::-webkit-search-results-button, .input::-webkit-search-results-decoration {
  display: none;
}

.input:hover {
  box-shadow: inset 0 0 0 1px #b3b3b3;
}

.input:focus, .input:active {
  outline: 0;
  box-shadow: inset 0 0 0 1px #3E82F7;
  background: #FFFFFF;
}

.input::placeholder {
  color: #AAAAAA;
}

.submit {
  position: absolute;
  top: 0;
  right: 0;
  left: inherit;
  margin: 0;
  border: 0;
  border-radius: 0 3px 3px 0;
  background-color: #3e82f7;
  padding: 0;
  width: 49px;
  height: 100%;
  vertical-align: middle;
  text-align: center;
  font-size: inherit;
  user-select: none;
}

.submit svg {
  width: 21px;
  height: 21px;
  vertical-align: middle;
  fill: #FFFFFF;
}

.submit::before {
  display: inline-block;
  margin-right: -4px;
  height: 100%;
  vertical-align: middle;
  content: '';
}

.reset {
  display: none;
  position: absolute;
  top: 10px;
  right: 56px;
  margin: 0;
  border: 0;
  background: none;
  cursor: pointer;
  padding: 0;
  font-size: inherit;
  user-select: none;
  fill: rgba(0, 0, 0, 0.5);
}

.reset svg {
  display: block;
  margin: 4px;
  width: 13px;
  height: 13px;
}
.input:valid ~ .reset {
  display: block;
  animation-name: sbx-reset-in;
  animation-duration: .15s;
}

@keyframes sbx-reset-in {
  0% {
    transform: translate3d(-20%, 0, 0);
    opacity: 0;
  }
  100% {
    transform: none;
    opacity: 1;
  }
}