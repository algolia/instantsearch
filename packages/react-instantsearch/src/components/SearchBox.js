import React, {Component, PropTypes} from 'react';
import translatable from '../core/translatable';
import classNames from './classNames.js';

const cx = classNames('SearchBox');

class SearchBox extends Component {
  static propTypes = {
    currentRefinement: PropTypes.string,
    refine: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,

    resetComponent: PropTypes.element,
    submitComponent: PropTypes.element,

    focusShortcuts: PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    ),

    autoFocus: PropTypes.bool,

    searchAsYouType: PropTypes.bool,
    onSubmit: PropTypes.func,
    onReset: PropTypes.func,
    onChange: PropTypes.func,

    // For testing purposes
    __inputRef: PropTypes.func,
  };

  static defaultProps = {
    currentRefinement: '',
    focusShortcuts: ['s', '/'],
    autoFocus: false,
    searchAsYouType: true,
  };

  constructor(props) {
    super();

    this.state = {
      query: props.searchAsYouType ? null : props.currentRefinement,
    };
  }

  componentDidMount() {
    document.addEventListener('keydown', this.onKeyDown);
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.onKeyDown);
  }

  componentWillReceiveProps(nextProps) {
    // Reset query when the searchParameters query has changed.
    // This is kind of an anti-pattern (props in state), but it works here
    // since we know for sure that searchParameters having changed means a
    // new search has been triggered.
    if (
      !nextProps.searchAsYouType &&
      nextProps.currentRefinement !== this.props.currentRefinement
    ) {
      this.setState({
        query: nextProps.currentRefinement,
      });
    }
  }

  getQuery = () =>
    this.props.searchAsYouType ?
      this.props.currentRefinement :
      this.state.query;

  setQuery = val => {
    const {refine, searchAsYouType} = this.props;
    if (searchAsYouType) {
      refine(val);
    } else {
      this.setState({
        query: val,
      });
    }
  }

  onInputMount = input => {
    this.input = input;
    if (this.props.__inputRef) {
      this.props.__inputRef(input);
    }
  };

  // From https://github.com/algolia/autocomplete.js/pull/86
  onKeyDown = e => {
    if (!this.props.focusShortcuts) {
      return;
    }

    const shortcuts = this.props.focusShortcuts.map(key =>
      typeof key === 'string' ? key.toUpperCase().charCodeAt(0) : key
    );

    const elt = e.target || e.srcElement;
    const tagName = elt.tagName;
    if (
      elt.isContentEditable ||
      tagName === 'INPUT' ||
      tagName === 'SELECT' ||
      tagName === 'TEXTAREA'
    ) {
      // already in an input
      return;
    }

    const which = e.which || e.keyCode;
    if (shortcuts.indexOf(which) === -1) {
      // not the right shortcut
      return;
    }

    this.input.focus();
    e.stopPropagation();
    e.preventDefault();
  };

  onSubmit = e => {
    e.preventDefault();
    e.stopPropagation();
    this.input.blur();

    const {refine, searchAsYouType} = this.props;
    if (!searchAsYouType) {
      refine(this.getQuery());
    }
    return false;
  };

  onChange = e => {
    this.setQuery(e.target.value);

    if (this.props.onChange) {
      this.props.onChange(e);
    }
  };

  onReset = () => {
    this.setQuery('');
    this.input.focus();

    if (this.props.onReset) {
      this.props.onReset();
    }
  };

  render() {
    const {
      translate,
      autoFocus,
    } = this.props;
    const query = this.getQuery();

    const submitComponent = this.props.submitComponent
      ? this.props.submitComponent
      : <svg role="img">
          <use xlinkHref="#sbx-icon-search-13" />
        </svg>;

    const resetComponent = this.props.resetComponent
      ? this.props.resetComponent
      : <svg role="img">
          <use xlinkHref="#sbx-icon-clear-3" />
        </svg>;

    const searchInputEvents = Object.keys(this.props).reduce((props, prop) => {
      if (
          ['onsubmit', 'onreset', 'onchange'].indexOf(prop.toLowerCase()) === -1 &&
          prop.indexOf('on') === 0
        ) {
        return {...props, [prop]: this.props[prop]};
      }

      return props;
    }, {});

    /* eslint-disable max-len */
    return (
      <form
        noValidate
        onSubmit={this.props.onSubmit ? this.props.onSubmit : this.onSubmit}
        onReset={this.onReset}
        {...cx('root')}
        action=""
        role="search"
      >
        <svg xmlns="http://www.w3.org/2000/svg" style={{display: 'none'}}>
          <symbol xmlns="http://www.w3.org/2000/svg" id="sbx-icon-search-13" viewBox="0 0 40 40">
            <path d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
            fillRule="evenodd" />
          </symbol>
          <symbol xmlns="http://www.w3.org/2000/svg" id="sbx-icon-clear-3" viewBox="0 0 20 20">
            <path d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z" fillRule="evenodd" />
          </symbol>
        </svg>
        <div
          role="search"
          {...cx('wrapper')}
        >
          <input
            ref={this.onInputMount}
            type="search"
            placeholder={translate('placeholder')}
            autoFocus={autoFocus}
            autoComplete="off"
            autoCorrect="off"
            autoCapitalize="off"
            spellCheck="false"
            required
            value={query}
            onChange={this.onChange}
            {...searchInputEvents}
            {...cx('input')}
          />
          <button type="submit" title={translate('submitTitle')} {...cx('submit')}>
            {submitComponent}
          </button>
          <button type="reset" title={translate('resetTitle')} {...cx('reset')} onClick={this.onReset}>
            {resetComponent}
          </button>
        </div>
      </form>
    );
    /* eslint-enable */
  }
}

export default translatable({
  resetTitle: 'Clear the search query.',
  submitTitle: 'Submit your search query.',
  placeholder: 'Search hereâ€¦',
})(SearchBox);
