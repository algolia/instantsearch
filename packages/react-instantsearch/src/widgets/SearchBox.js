import connectSearchBox from '../connectors/connectSearchBox.js';
import SearchBoxComponent from '../components/SearchBox.js';

/**
 * The SearchBox component displays a search box that lets the user search for a specific query.
 * @name SearchBox
 * @kind widget
 * @propType {string[]} [focusShortcuts=['s','/']] - List of keyboard shortcuts that focus the search box. Accepts key names and key codes.
 * @propType {boolean} [autoFocus=false] - Should the search box be focused on render?
 * @propType {boolean} [searchAsYouType=true] - Should we search on every change to the query? If you disable this option, new searches will only be triggered by clicking the search button or by pressing the enter key while the search box is focused.
 * @propType {function} [onSubmit] - Intercept submit event sent from the SearchBox form container.
 * @propType {function} [onReset] - Listen to `reset` event sent from the SearchBox form container.
 * @propType {function} [on*] - Listen to any events sent form the search input itself.
 * @propType {React.Element} [submitComponent] - Change the apparence of the default submit button (magnifying glass).
 * @propType {React.Element} [resetComponent] - Change the apparence of the default reset button (cross).
 * @propType {string} [defaultRefinement] - Provide default refinement value when component is mounted.
 * @themeKey ais-SearchBox__root - the root of the component
 * @themeKey ais-SearchBox__wrapper - the search box wrapper
 * @themeKey ais-SearchBox__input - the search box input
 * @themeKey ais-SearchBox__submit - the submit button
 * @themeKey ais-SearchBox__reset - the reset button
 * @translationkey submitTitle - The submit button title
 * @translationkey resetTitle - The reset button title
 * @translationkey placeholder - The label of the input placeholder
 * @example
 * import React from 'react';
 *
 * import {SearchBox, InstantSearch} from '../packages/react-instantsearch/dom';
 *
 * export default function App() {
 *   return (
 *     <InstantSearch
 *       appId="latency"
 *       apiKey="6be0576ff61c053d5f9a3225e2a90f76"
 *       indexName="ikea"
 *     >
 *       <SearchBox />
 *     </InstantSearch>
 *   );
 * }
 */
export default connectSearchBox(SearchBoxComponent);
