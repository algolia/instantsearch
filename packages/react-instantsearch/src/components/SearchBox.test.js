/* eslint-env jest, jasmine */

import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

import SearchBox from './SearchBox';

describe('SearchBox', () => {
  it('applies its default props', () => {
    const tree = renderer.create(
      <SearchBox refine={() => null} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('transfers the autoFocus prop to the underlying input element', () => {
    const tree = renderer.create(
      <SearchBox refine={() => null} autoFocus />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('treats its query prop as its input value', () => {
    const inst = renderer.create(
      <SearchBox refine={() => null} currentRefinement="QUERY1" />
    );
    expect(inst.toJSON()).toMatchSnapshot();

    inst.update(
      <SearchBox refine={() => null} currentRefinement="QUERY2" />
    );
    expect(inst.toJSON()).toMatchSnapshot();
  });
  it('lets you customize its theme', () => {
    const tree = renderer.create(
      <SearchBox
        refine={() => null}
        theme={{
          root: 'ROOT',
          wrapper: 'WRAPPER',
          input: 'INPUT',
          submit: 'SUBMIT',
          reset: 'RESET',
        }}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('lets you customize its translations', () => {
    const tree = renderer.create(
      <SearchBox
        refine={() => null}
        translations={{
          submit: 'SUBMIT',
          reset: 'RESET',
          resetTitle: 'RESET_TITLE',
          placeholder: 'PLACEHOLDER',
        }}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('treats query as a default value when searchAsYouType=false', () => {
    const wrapper = mount(
        <SearchBox refine={() => null} currentRefinement="QUERY1" searchAsYouType={false} />
      );
    expect(wrapper.find('input').props().value).toBe('QUERY1');
    wrapper.find('input').simulate('change', {target: {value: 'QUERY2'}});
    expect(wrapper.find('input').props().value).toBe('QUERY2');
    wrapper.unmount();
  });

  it('refines its value on change when searchAsYouType=true', () => {
    const refine = jest.fn();
    const wrapper = mount(
        <SearchBox
          searchAsYouType
          refine={refine}
        />
      );
    wrapper.find('input').simulate('change', {target: {value: 'hello'}});
    expect(refine.mock.calls.length).toBe(1);
    expect(refine.mock.calls[0][0]).toBe('hello');
    wrapper.unmount();
  });

  it('only refines its query on submit when searchAsYouType=false', () => {
    const refine = jest.fn();
    const wrapper = mount(
        <SearchBox
          searchAsYouType={false}
          refine={refine}
        />
      );
    wrapper.find('input').simulate('change', {target: {value: 'hello'}});
    expect(refine.mock.calls.length).toBe(0);
    wrapper.find('form').simulate('submit');
    expect(refine.mock.calls.length).toBe(1);
    expect(refine.mock.calls[0][0]).toBe('hello');
    wrapper.unmount();
  });

  it('onSubmit behavior should be override if provided as props', () => {
    const onSubmit = jest.fn();
    const refine = jest.fn();
    const wrapper = mount(
      <SearchBox
        searchAsYouType={false}
        onSubmit={onSubmit}
        refine={refine}
      />
    );
    wrapper.find('form').simulate('submit');
    expect(onSubmit.mock.calls.length).toBe(1);
    expect(refine.mock.calls.length).toBe(0);
    wrapper.unmount();
  });

  it('focuses the input when one of the keys in focusShortcuts is pressed', () => {
    let input;
    mount(
      <SearchBox
        refine={() => null}
        focusShortcuts={['s', 84]}
        __inputRef={node => {
          input = node;
        }}
      />
    );

    input.focus = jest.fn();
    const event1 = new KeyboardEvent('keydown', {keyCode: 82});
    document.dispatchEvent(event1);
    expect(input.focus.mock.calls.length).toBe(0);
    const event2 = new KeyboardEvent('keydown', {keyCode: 83});
    document.dispatchEvent(event2);
    expect(input.focus.mock.calls.length).toBe(1);
    const event3 = new KeyboardEvent('keydown', {keyCode: 84});
    document.dispatchEvent(event3);
    expect(input.focus.mock.calls.length).toBe(2);
  });
});
