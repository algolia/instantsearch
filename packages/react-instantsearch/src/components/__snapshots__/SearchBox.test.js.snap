// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox applies its default props 1`] = `
<form
  action=""
  className="ais-SearchBox__root"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
  role="search"
>
  <svg
    style={
      Object {
        "display": "none",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <symbol
      id="sbx-icon-search-13"
      viewBox="0 0 40 40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
        fillRule="evenodd"
      />
    </symbol>
    <symbol
      id="sbx-icon-clear-3"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
        fillRule="evenodd"
      />
    </symbol>
  </svg>
  <div
    className="ais-SearchBox__wrapper"
    role="search"
  >
    <input
      autoCapitalize="off"
      autoComplete="off"
      autoCorrect="off"
      autoFocus={false}
      className="ais-SearchBox__input"
      onChange={[Function]}
      placeholder="Search here‚Ä¶"
      required={true}
      spellCheck="false"
      type="search"
      value=""
    />
    <button
      className="ais-SearchBox__submit"
      title="Submit your search query."
      type="submit"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-search-13"
        />
      </svg>
    </button>
    <button
      className="ais-SearchBox__reset"
      onClick={[Function]}
      title="Clear the search query."
      type="reset"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-clear-3"
        />
      </svg>
    </button>
  </div>
</form>
`;

exports[`SearchBox lets you customize its theme 1`] = `
<form
  action=""
  className="ais-SearchBox__root"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
  role="search"
>
  <svg
    style={
      Object {
        "display": "none",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <symbol
      id="sbx-icon-search-13"
      viewBox="0 0 40 40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
        fillRule="evenodd"
      />
    </symbol>
    <symbol
      id="sbx-icon-clear-3"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
        fillRule="evenodd"
      />
    </symbol>
  </svg>
  <div
    className="ais-SearchBox__wrapper"
    role="search"
  >
    <input
      autoCapitalize="off"
      autoComplete="off"
      autoCorrect="off"
      autoFocus={false}
      className="ais-SearchBox__input"
      onChange={[Function]}
      placeholder="Search here‚Ä¶"
      required={true}
      spellCheck="false"
      type="search"
      value=""
    />
    <button
      className="ais-SearchBox__submit"
      title="Submit your search query."
      type="submit"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-search-13"
        />
      </svg>
    </button>
    <button
      className="ais-SearchBox__reset"
      onClick={[Function]}
      title="Clear the search query."
      type="reset"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-clear-3"
        />
      </svg>
    </button>
  </div>
</form>
`;

exports[`SearchBox lets you customize its translations 1`] = `
<form
  action=""
  className="ais-SearchBox__root"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
  role="search"
>
  <svg
    style={
      Object {
        "display": "none",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <symbol
      id="sbx-icon-search-13"
      viewBox="0 0 40 40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
        fillRule="evenodd"
      />
    </symbol>
    <symbol
      id="sbx-icon-clear-3"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
        fillRule="evenodd"
      />
    </symbol>
  </svg>
  <div
    className="ais-SearchBox__wrapper"
    role="search"
  >
    <input
      autoCapitalize="off"
      autoComplete="off"
      autoCorrect="off"
      autoFocus={false}
      className="ais-SearchBox__input"
      onChange={[Function]}
      placeholder="PLACEHOLDER"
      required={true}
      spellCheck="false"
      type="search"
      value=""
    />
    <button
      className="ais-SearchBox__submit"
      title="Submit your search query."
      type="submit"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-search-13"
        />
      </svg>
    </button>
    <button
      className="ais-SearchBox__reset"
      onClick={[Function]}
      title="RESET_TITLE"
      type="reset"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-clear-3"
        />
      </svg>
    </button>
  </div>
</form>
`;

exports[`SearchBox lets you give custom components for reset and submit 1`] = `
<form
  action=""
  className="ais-SearchBox__root"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
  role="search"
>
  <svg
    style={
      Object {
        "display": "none",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <symbol
      id="sbx-icon-search-13"
      viewBox="0 0 40 40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
        fillRule="evenodd"
      />
    </symbol>
    <symbol
      id="sbx-icon-clear-3"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
        fillRule="evenodd"
      />
    </symbol>
  </svg>
  <div
    className="ais-SearchBox__wrapper"
    role="search"
  >
    <input
      autoCapitalize="off"
      autoComplete="off"
      autoCorrect="off"
      autoFocus={false}
      className="ais-SearchBox__input"
      onChange={[Function]}
      placeholder="Search here‚Ä¶"
      required={true}
      spellCheck="false"
      type="search"
      value=""
    />
    <button
      className="ais-SearchBox__submit"
      title="Submit your search query."
      type="submit"
    >
      <span>
        üîç
      </span>
    </button>
    <button
      className="ais-SearchBox__reset"
      onClick={[Function]}
      title="Clear the search query."
      type="reset"
    >
      <svg
        viewBox="200 198 108 122"
      >
        <path
          d="M200.8 220l45 46.7-20 47.4 31.7-34 50.4 39.3-34.3-52.6 30.2-68.3-49.7 51.7"
        />
      </svg>
    </button>
  </div>
</form>
`;

exports[`SearchBox transfers the autoFocus prop to the underlying input element 1`] = `
<form
  action=""
  className="ais-SearchBox__root"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
  role="search"
>
  <svg
    style={
      Object {
        "display": "none",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <symbol
      id="sbx-icon-search-13"
      viewBox="0 0 40 40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
        fillRule="evenodd"
      />
    </symbol>
    <symbol
      id="sbx-icon-clear-3"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
        fillRule="evenodd"
      />
    </symbol>
  </svg>
  <div
    className="ais-SearchBox__wrapper"
    role="search"
  >
    <input
      autoCapitalize="off"
      autoComplete="off"
      autoCorrect="off"
      autoFocus={true}
      className="ais-SearchBox__input"
      onChange={[Function]}
      placeholder="Search here‚Ä¶"
      required={true}
      spellCheck="false"
      type="search"
      value=""
    />
    <button
      className="ais-SearchBox__submit"
      title="Submit your search query."
      type="submit"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-search-13"
        />
      </svg>
    </button>
    <button
      className="ais-SearchBox__reset"
      onClick={[Function]}
      title="Clear the search query."
      type="reset"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-clear-3"
        />
      </svg>
    </button>
  </div>
</form>
`;

exports[`SearchBox treats its query prop as its input value 1`] = `
<form
  action=""
  className="ais-SearchBox__root"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
  role="search"
>
  <svg
    style={
      Object {
        "display": "none",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <symbol
      id="sbx-icon-search-13"
      viewBox="0 0 40 40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
        fillRule="evenodd"
      />
    </symbol>
    <symbol
      id="sbx-icon-clear-3"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
        fillRule="evenodd"
      />
    </symbol>
  </svg>
  <div
    className="ais-SearchBox__wrapper"
    role="search"
  >
    <input
      autoCapitalize="off"
      autoComplete="off"
      autoCorrect="off"
      autoFocus={false}
      className="ais-SearchBox__input"
      onChange={[Function]}
      placeholder="Search here‚Ä¶"
      required={true}
      spellCheck="false"
      type="search"
      value="QUERY1"
    />
    <button
      className="ais-SearchBox__submit"
      title="Submit your search query."
      type="submit"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-search-13"
        />
      </svg>
    </button>
    <button
      className="ais-SearchBox__reset"
      onClick={[Function]}
      title="Clear the search query."
      type="reset"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-clear-3"
        />
      </svg>
    </button>
  </div>
</form>
`;

exports[`SearchBox treats its query prop as its input value 2`] = `
<form
  action=""
  className="ais-SearchBox__root"
  noValidate={true}
  onReset={[Function]}
  onSubmit={[Function]}
  role="search"
>
  <svg
    style={
      Object {
        "display": "none",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <symbol
      id="sbx-icon-search-13"
      viewBox="0 0 40 40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
        fillRule="evenodd"
      />
    </symbol>
    <symbol
      id="sbx-icon-clear-3"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
        fillRule="evenodd"
      />
    </symbol>
  </svg>
  <div
    className="ais-SearchBox__wrapper"
    role="search"
  >
    <input
      autoCapitalize="off"
      autoComplete="off"
      autoCorrect="off"
      autoFocus={false}
      className="ais-SearchBox__input"
      onChange={[Function]}
      placeholder="Search here‚Ä¶"
      required={true}
      spellCheck="false"
      type="search"
      value="QUERY2"
    />
    <button
      className="ais-SearchBox__submit"
      title="Submit your search query."
      type="submit"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-search-13"
        />
      </svg>
    </button>
    <button
      className="ais-SearchBox__reset"
      onClick={[Function]}
      title="Clear the search query."
      type="reset"
    >
      <svg
        role="img"
      >
        <use
          xlinkHref="#sbx-icon-clear-3"
        />
      </svg>
    </button>
  </div>
</form>
`;
